src/sortStudents.ts

```diff
@@ -1,16 +1,63 @@
-
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const people: Student[] = [...students];
+
+  people.sort((personA, personB) => {
+    let personAAverage: number;
+    let personBAverage: number;
+
+    switch (sortBy) {
+      case SortType.Name:
+      case SortType.Surname:
+        return order === 'desc'
+          ? personB[sortBy].localeCompare(personA[sortBy])
+          : personA[sortBy].localeCompare(personB[sortBy]);
+
+      case SortType.Age:
+      case SortType.Married:
+        return order === 'desc'
+          ? +personB[sortBy] - +personA[sortBy]
```

Comment:
Try to use Number() constuctor, it's more obvious and noticeable in code than operator `+`
```suggestion
          ? Number(personB[sortBy]) - +personA[sortBy]
```

Reply:
Number was implemented